.github/workflows/compound-java-dev.yml

```diff
@@ -106,9 +124,12 @@ jobs:
 
 # if Sonatype
   sonatype-push:
-    if: (contains(inputs.build-target, 'Sonatype') || contains(inputs.build-target, 'sonatype')) && (inputs.strict-scanner && success() || !inputs.strict-scanner )
+    if: |
+      (contains(inputs.build-target, 'Sonatype') || contains(inputs.build-target, 'sonatype')) && !cancelled() &&
+      (inputs.strict-scanner && needs.license-check.result == 'success' && needs.owasp-scan.result == 'success' || !inputs.strict-scanner) &&
+      (inputs.integration-test-enabled && needs.integration-test.result == 'success' || !inputs.integration-test-enabled)
```

Comment:
suggestion: To be honest, it looks terrifying right now... What if we create a new job that will compute the part that is related to `strict-scanner` and `integration-test`? In this case, we could reuse the job result in both `sonatype-push` and `docker-push` jobs

Reply:
Do you mean a separate compound workflow ?

Reply:
No, I meant creating a job in compound-java and compound-java-dev that will calculate the result of checks (owasp, license and integration test). But if you think that extracting that into a separate workflow would be better - you can do so

Reply:
Checked the next cases:
- scanner-enabled: false, strict-scanner: false, integration-test-enabled: false 
- scanner-enabled: false, strict-scanner: true, integration-test-enabled: false 
- scanner-enabled: true, strict-scanner: true, integration-test-enabled: false 
- scanner-enabled: true, strict-scanner: true, integration-test-enabled: true https://github.com/th2-net/th2-cradle-admin-tool/actions/runs/12393396689